main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -336,8 +336,19 @@ object GenExpression {
       visitor.visitLabel(afterFinally)
 
     case Expression.ScopeExit(exp1, exp2, tpe, loc) =>
-      //!TODO: For now, just emit unit
-      compileConstant(visitor, Ast.Constant.Unit, MonoType.Unit, loc)
+      // Compile the expression representing the region
+      compileExpression(exp2, visitor, currentClass, lenv0, entryPoint)
```

Comment:
I don't know whats assumed about `exp1` and `exp2` here, but its okay do run `exp2` before `exp1`? Shouldn't we adhere to left-to-right evaluation?

Reply:
I think they should be evaluated exp1 then exp and use a swap to rearrange them.

Reply:
We should always be left-to-right.

Reply:
Ah, sorry, didn't realise that this was a requirement. Will fix.